### PR TITLE
Remove deprecated compatibility layer

### LINKED ISSUE #1393: Cleanup AsyncRestTemplate support
## Context
AsyncRestTemplate is removed from spring framework 6.

## Possible Implementation
* Remove the support from riptide-compatibility mode
* Record this change in the MIGRATION.md in new section - migration to version 4.0

### PR DESCRIPTION
**Pull request by dingxiangfei2009**
## Description

Fix #1393 

This PR removes `AsyncHttpOperations` in preparation of Riptide 4.

_Question_ do we still need to keep `riptide-compatibility`?

## Motivation and Context

`RestTemplate` and `AsyncRestTemplate` are deprecated in Spring 4 and removed in Spring 6. Because of this, we are removing the facades from the compatibility mode.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by fatroom** [comment]
@dingxiangfei2009 resttemplate is not deprecated: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html so we still need this module. It's only AsyncRestTemplate deprecated and removed

File: MIGRATION.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+# Riptide 4.0 Migration Guide
+
+## Breaking Changes
+
+Since Spring 5, `RestTemplate` and `AsyncRestTemplate` are deprecated in favor of `WebClient`.
```
Comment (fatroom) [inline MIGRATION.md:5]: Only `AsyncRestTemplate` deprecated in favour `WebClient`

Comment (dingxiangfei2009) [reply MIGRATION.md]: Sorry. The relevant classes are restored.

**Comment by fatroom** [comment]
@dingxiangfei2009 thank you for your contribution!

**Comment by fatroom** [comment]
👍

**Pull request merged**